valhalla/baldr/graphreader.h

```diff
@@ -157,6 +157,7 @@ class FlatTileCache : public TileCache {
   }
   inline uint32_t get_index(const GraphId& graphid) const {
     auto offset = get_offset(graphid);
+    // TODO a negative value can not be stored on an uint32_t, anything can happen
```

Comment:
we do this a lot in the code making use of automatic conversion to set max value by wrapping around with the negative value on an unsigned type. we often use the max value as a sentinel value on unsigned types to signify invalid

Reply:
Ok, so I changed to `std::numeric_limits<uint32_t>::max();`

Reply:
i feel like that hurts readibilty compared to `-1`, maybe we should make something like

```c++
template <typename numeric_t> invalid() { return std::numeric_limits<numeric_t>::max(); }
```

then in calling code at least it would read nicely:

```c++
return offset < cache_indices_.size() ? cache_indices_[offset] : invalid<uint32_t>();
```

Reply:
Added this change to the list of tasks, on #4018 
When encountering with this again, need to figure out the best place to put the template.
Will do that on a later PR

Reply:
ah, I just saw the comment of where to place it